outputs/openapi/generate.go

```diff
@@ -124,23 +128,21 @@ func Generate(filePath string) astra.ServiceFunction {
 			for _, returnType := range endpoint.ReturnTypes {
 				s.Log.Debug().Str("endpointPath", endpoint.Path).Str("method", endpoint.Method).Str("return", returnType.Field.Name).Msg("Adding return type")
 				var mediaType MediaType
-				bindingType := astra.ContentTypeToBindingTag(endpoint.ContentType)
+				bindingType := astra.ContentTypeToBindingTag(returnType.ContentType)
 				mediaType.Schema = mapFieldToSchema(bindingType, returnType.Field)
 
-				var content map[string]MediaType
-				if mediaType.Schema.Type != "" || mediaType.Schema.Ref != "" {
-					content = map[string]MediaType{
-						endpoint.ContentType: mediaType,
+				statusCode := strconv.Itoa(returnType.StatusCode)
+				if _, set := operation.Responses[statusCode]; !set {
+					operation.Responses[statusCode] = Response{
+						Description: "",
```

Comment:
Nit: if empty do we need to define them here? Mainly for `Links` and `Description`, completely see the argument for `Content` as an empty map is nil

Reply:
I see what you saying but even if I initialise the `Content` map here but don't add any items it's still considered empty. I problem I was having before was that I was putting empty structs into the map.

Reply:
It's a nit pick, doesn't affect functionality at all